src/backends/r1cs/mod.rs

```diff
@@ -1,3 +1,7 @@
+// TODO: There is a bunch of places where there are unused vars.
+// Remove this lint allowance when fixed.
+#![allow(unused_variables)]
```

Comment:
imo this is dangerous, as these unused variables might be bugs and we should address that instead of silencing it

Reply:
Created #101

Reply:
fixed in #105